sihl/src/contract_email.ml

```diff
@@ -9,6 +9,9 @@ type t =
   }
 
 let name = "email"
+let name_smtp = Format.asprintf "%s_smtp" name
```

Comment:
@aronerben I suspect we need to create a unique name in Sihl when multiple email services are used. Imo the name doesn't need to contain the implementation, so something like `email-0`, `email-1` could work.

Reply:
@jerben You're right. The services don't know of each others existence when they are being registered. I see two options to avoid naming conflicts:
1. The user passes in a unique name as an arg for `Email.register`
2. `Core_container` does a preprocessing step where same named services are renamed to the names you proposed

I would say 2. is preferable here, because the user should not be responsible for this. What do you think?

Reply:
I agree, but since you can't trust users to actually provide unique names we need to implement 2) anyway to just silently "do the right thing". 1) could be the next step.

Reply:
This is tougher to do than I expected. If you adjust the naming for duplicated services, you also have to adjust their names when they are listed as dependencies...

So for example a queued email depends on an smtp mail and the smtp mail is also registered. This means you first have to resolve the name conflict to something like `queued: email-0`, `smtp: email-1` and now you have to find all dependencies on this specific `smtp` mail and rename the name to `email-1` there too. I don't see a clear association to do that. Do you have an idea?

Reply:
@aronerben 
You are right. 
As discussed, maybe something like that could solve the issue:
```ocaml
let counter = ref 0

let create_lifecycle
    ?(dependencies = fun () -> [])
    ?(start = fun () -> Lwt.return ())
    ?(stop = fun () -> Lwt.return ())
    ?implementation_name
    name
  =
  counter := !counter + 1;
  { type_name = name
  ; implementation_name
  ; id = !counter
  ; dependencies
  ; start
  ; stop
  }
;;
```

Reply:
Done